scripts/release-notes/index.js

```diff
@@ -1,30 +1,43 @@
-/*
- * Generates a changelog given a GH report and milestone.
- *
- * USAGE:
- *  1) Make sure all the issues in the release are assigned to the milestone
- *  2) Each issue should have one and only one Type label - the script will tell you which ones don't
- *  3) Make sure you have generated a GH token and created a token.json file, eg: { "githubApiToken": "..." }
- *     This token needs at least `read:org` permissions
- *  4) Execute the command: node index.js <repo_name> <milestone_name> > <output_file>
- *      eg: node index.js cht-core 3.0.0 > tmp.md
- *  5) Insert the contents of the output file into the appropriate location in Changes.md
- */
-
+const minimist = require('minimist');
 const { Octokit } = require('@octokit/rest');
+const { graphql } = require('@octokit/graphql');
 const octokit = new Octokit({
   auth: require('../token.json').githubApiToken,
   userAgent: 'cht-release-note-generator',
 });
+const authenticatedGraphql = graphql.defaults({
+  headers: {
+    authorization: `token ${require('../token.json').githubApiToken}`,
+    userAgent: 'cht-release-note-generator',
+  },
+});
```

Comment:
It would be interesting to replace REST API calls so that we have only one way of interacting with GitHub's API

There are only the two calls on [line 203](https://github.com/medic/cht-core/blob/8987_validate_commit_milestones/scripts/release-notes/index.js#L203) and [line 237](https://github.com/medic/cht-core/blob/8987_validate_commit_milestones/scripts/release-notes/index.js#L237).  
I looked into them and we could replace both of them with a single gql query like this:
```js
const getMilestoneIssues = async () => {
  const response = await gitHubRepoQuery(`
    milestones(query: "${MILESTONE_NAME}", first: 1) {
      nodes {
        issues(first: 100, states: [OPEN, CLOSED]) {
          nodes {
            number
            url
            title
            labels(first: 100) {
              nodes {
                name
              }
            }
          }
        }
      }
    }
  `);
  return response.repository.milestones.nodes[0].issues.nodes;
};
```

The only thing I don't like about this solution is the lack of pagination handling. GitHub provides [this utility package](https://www.npmjs.com/package/@octokit/plugin-paginate-graphql) to have the same pagination DX their REST client has with `octokit.paginate(route, params)`

Reply:
Turns out nested pagination (and apparently querying `issues` inside of `milestones` counts as nested pagination) is frowned upon and [not supported](https://github.com/octokit/plugin-paginate-graphql.js/tree/v4.0.1?tab=readme-ov-file#unsupported-nested-pagination) by `plugin-paginate-graphql`, so I had to keep an extra query to get the milestone number, but otherwise this works great!